src/get-options-overrides.ts

```diff
@@ -16,13 +16,26 @@ export function getOptionsOverrides({ useTsconfigDeclarationDir, cacheRoot }: IO
 		noEmitOnError: false,
 		inlineSourceMap: false,
 		outDir: normalize(`${cacheRoot}/placeholder`), // need an outdir that is different from source or tsconfig parsing trips up. https://github.com/Microsoft/TypeScript/issues/24715
-		moduleResolution: tsModule.ModuleResolutionKind.NodeJs,
 		allowNonTsExtensions: true,
+		moduleResolution: tsModule.ModuleResolutionKind.Node10,
 	};
 
 	if (!preParsedTsconfig)
 		return overrides;
 
+	switch (preParsedTsconfig.options.moduleResolution)
+	{
+		case tsModule.ModuleResolutionKind.Node10:
+		case tsModule.ModuleResolutionKind.Node16:
+		case tsModule.ModuleResolutionKind.NodeNext:
+			overrides.moduleResolution = preParsedTsconfig.options.moduleResolution;
+			break;
+		case tsModule.ModuleResolutionKind.Classic:
+		case tsModule.ModuleResolutionKind.Bundler:
+		default:
+			overrides.moduleResolution = tsModule.ModuleResolutionKind.Node10;
+	}
+
```

Comment:
with no override above, could simplify this to:
```suggestion
	if (preParsedTsconfig.options.moduleResolution === tsModule.ModuleResolutionKind.Classic)
	  overrides.moduleResolution = tsModule.ModuleResolutionKind.Node10;
```